ios-target-app/Home/Views/HomeViewController.swift

```diff
@@ -105,15 +112,32 @@ class HomeViewController: UIViewController, MKMapViewDelegate {
   }
   
   private func setMapConstraints() {
-    view.addSubview(mapView)
+    view.addSubviews([mapView, createTargetBottomSheetView])
     
     mapView.translatesAutoresizingMaskIntoConstraints = false
     
+    createTargetBottomSheetView.attachHorizontally(to: view, leadingMargin: 0, trailingMargin: 0)
+    
     NSLayoutConstraint.activate([
       mapView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
       mapView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
       mapView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
-      mapView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor)
+      mapView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+      createTargetBottomSheetView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+      createTargetBottomSheetView.heightAnchor.constraint(equalToConstant: UI.BottomSheet.height)
     ])
   }
 }
+
+extension HomeViewController: BottomSheetDelegate {
+  func createTargetButtonTapped() {
+    let saveTargetBottomSheetViewController = SaveTargetBottomSheetViewController()
+    let navigationController = UINavigationController(rootViewController: saveTargetBottomSheetViewController)
```

Comment:
do you really need to embed the view controller into a navigation view controller?